boofuzz/blocks/block.py

```diff
@@ -68,7 +67,7 @@ def original_value(self):
     def name(self):
         return self._name
 
-    def mutate(self):
+    def mutate(self):  # noqa: C901
```

Comment:
IMO "function is too complex" is a legitimate observation here. I'm working on another branch (mutations-iterator) on some changes that may end up helping.

Reply:
Good to hear, I'll remove that ignore then.

Right now the max complexity is set to 15 and the mutate function is the only one with an even higher complexity.
The default would be 10, which in our case is violated by quite a few functions (e.g. sessions/init).

> This feature is quite useful to detect over-complex code. According to McCabe, anything that goes beyond 10 is too complex. See https://en.wikipedia.org/wiki/Cyclomatic_complexity.
https://pypi.org/project/flake8/2.4.1/

If you want to be more strict about the code complexity we can turn it down to the default, but it's going to be hard to follow imo.

Here is a list of all functions that violate the default complexity:
```python
./network_monitor.py:247:1: C901 'main' is too complex (12)
./vmcontrol.py:40:5: C901 'VMControlPedrpcServer.__init__' is too complex (15)
./vmcontrol.py:514:1: C901 'If 514' is too complex (15)
./utils/ida_fuzz_library_extender.py:90:1: C901 'find_strings' is too complex (11)
./boofuzz/sessions.py:323:5: C901 'Session.__init__' is too complex (12)
./boofuzz/sessions.py:1036:5: C901 'Session.transmit_normal' is too complex (15)
./boofuzz/sessions.py:1091:5: C901 'Session.transmit_fuzz' is too complex (13)
./boofuzz/socket_connection.py:119:5: C901 'SocketConnection.open' is too complex (15)
./boofuzz/socket_connection.py:176:5: C901 'SocketConnection.recv' is too complex (11)
./boofuzz/socket_connection.py:220:5: C901 'SocketConnection.send' is too complex (15)
./boofuzz/blocks/checksum.py:126:5: C901 'Checksum._checksum' is too complex (11)
./boofuzz/blocks/block.py:189:5: C901 'Block.render' is too complex (14)
```